### DISCUSSION TITLE
Is there a function to bulk manually integrate peaks with certain mz and rt

### DISCUSSION THREAD

**Discussion by h2002766** [shuzhao-li/asari#90] [Ideas]
Is there a function to bulk manually integrate peaks with certain mz range and rt range with Asari. This would be quite useful for gap filling in case Asari misses anything! A function similar to below one manualChromPeaks from XCMS would be useful to have within Asari

https://rdrr.io/github/sneumann/xcms/man/manualChromPeaks.html

**Comment by jmmitc06** [comment]
Unfortunately there is not. There is an issue open for the pipeline project based on asari to accomplish this. In future versions, I plan to implement this feature.

If possible, can you elaborate on your use case? Are you curious about this feature in abstract sense or have you found an example where this functionality is needed?

Mar 25, 2024 4:27:39 PM h2002766 ***@***.***>:

**Reply by h2002766** [reply]
We are seeing some missingness of peak area data despite being very good peaks and the gapfilling. RT alignment was also good. Hence, its good to have a manual peak area extract function using specific mz and rt windows similar to xcms manualChromPeaks function.

Also, why are peak areas completely different in Asari vs XCMS for any given metabolite?

**Reply by shuzhao-li** [reply]
Replied below. 
Peak areas are all relative values anyways. The numbers are different from how area integration is done.

**Comment by shuzhao-li** [comment]
This will be covered by "targeted extraction" when it's done.

**Reply by h2002766** [reply]
Targeted extraction is only from already peak picked data. I think it does not addresses missed peak picking. Correct me if I am wrong!

**Reply by shuzhao-li** [reply]
If your problem is split ion chromatograms, relaxing ppm in mass resolution should fix it.
Targeted extraction is to retrieve signals from a designated region (m/z and RT window), not dependent on peak detection (picking).
